test/python/transpiler/test_swap_cx_swap_to_bridge.py

```diff
@@ -0,0 +1,269 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Tests for the SwapCXSwapToBridge transpiler pass.
+"""
+
+import unittest
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.classicalregister import ClassicalRegister
+from qiskit.circuit.library import BridgeGate
+from qiskit.circuit.library.standard_gates import CXGate, SwapGate
+from qiskit.circuit.quantumregister import QuantumRegister
+from qiskit.transpiler import PassManager
+from qiskit.transpiler.passes.optimization import SwapCXSwapToBridge
+from qiskit.test import QiskitTestCase
+
+
+class TestSwapCXSwapToBridge(QiskitTestCase):
+    """
+    Tests to verify that blocks of Swap-CX-Swap are found and replaced correctly.
+    """
+
+    def test_one_occurence(self):
+        """Test with input circuit:
+        q_0: ─X───────X─
+              │       │
+        q_1: ─X───■───X─
+                ┌─┴─┐
+        q_2: ───┤ X ├───
+                └───┘
+        """
+        qc = QuantumCircuit(3)
+        qc.swap(0, 1)
+        qc.cx(1, 2)
+        qc.swap(0, 1)
+
+        pm = PassManager(SwapCXSwapToBridge())
+        circuit = pm.run(qc)
+
+        expected = QuantumCircuit(3)
+        expected.append(BridgeGate(3), [0, 1, 2])
+        self.assertEqual(circuit, expected)
+
+    def test_one_occurence_cx_reversed_direction(self):
+        """Test with input circuit:
+        q_0: ─X───────X─
+              │ ┌───┐ │
+        q_1: ─X─┤ X ├─X─
+                └─┬─┘
+        q_2: ─────■─────
```

Comment:
can you explain a bit how this is a bridge on [2, 1, 0]?

Reply:
Well, both circuits compute a CNOT with `q_2` as a control and `q_0` as a target.